src/acp/policy.ts

```diff
@@ -177,6 +178,48 @@ export function setPolicy<Dataset extends SolidDataset>(
   return setThing(policyResource, policy);
 }
 
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Given a [[Policy]] and a set of [[AccessModes]], return a new Policy based on the given
+ * Policy, but with the given Access Modes allowed on it.
+ *
+ * @param policy The Policy on which to set the modes to allow.
+ * @param modes Modes to allow for this Policy.
+ * @since Not released yet.
+ */
+export function setAllowModesV2<P extends Policy | ResourcePolicy>(
+  policy: P,
+  modes: AccessModes
+): P {
+  let newPolicy = removeAll(policy, acp.allow);
+
+  if (modes.read === true) {
+    newPolicy = addIri(
+      newPolicy,
+      acp.allow,
+      internal_accessModeIriStrings.read
```

Comment:
Just to make sure I get this straight, the access modes changed to be compatible with ACL, but the rest of the ACP vocabulary/spec is unchanged ?

Reply:
Yep! Had to check a couple of times during implementation myself as well.